File: aura-backend/server/models/Order.js
const mongoose = require('mongoose');

const orderItemSchema = new mongoose.Schema({
  product: { type: mongoose.Schema.Types.ObjectId, ref: 'Product' },
  name: { type: String, required: true },
  price: { type: Number, required: true },
  qty: { type: Number, required: true, min: 1 },
  serialNumber: { type: String, default: null },
});

const requestStateSchema = new mongoose.Schema(
  {
    status: {
      type: String,
      enum: ['none', 'pending', 'approved', 'rejected'],
      default: 'none',
    },
    reason: { type: String, default: '' },
    requestedAt: { type: Date, default: null },
    resolvedAt: { type: Date, default: null },
    resolvedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'Admin', default: null },
    note: { type: String, default: '' },
  },
  { _id: false }
);

const orderSchema = new mongoose.Schema(
  {
    orderNumber: { type: String, required: true, unique: true },
    user: { type: mongoose.Schema.Types.ObjectId, ref: 'User', default: null },
    items: [orderItemSchema],
    shippingAddress: {
      fullName: String,
      phone: String,
      email: String,
      address: String,
      city: String,
      district: String,
      ward: String,
      note: String,
    },
    status: {
      type: String,
      enum: ['pending', 'confirmed', 'processing', 'shipped', 'delivered', 'cancelled'],
      default: 'pending',
    },
    total: { type: Number, required: true },
    shippingFee: { type: Number, default: 0 },
    discount: { type: Number, default: 0 },
    appliedPromotion: {
      code: { type: String },
      discountPercent: { type: Number },
      discountAmount: { type: Number },
    },
    paymentMethod: {
      type: String,
      enum: ['cod', 'qr', 'momo', 'zalopay', 'atm', 'vietqr'],
      default: 'cod',
    },
    isPaid: { type: Boolean, default: false },
    paidAt: { type: Date, default: null },
    zaloPayTransId: { type: String, default: null },
    vietqrTransId: { type: String, default: null },
    shippedAt: { type: Date, default: null },
    deliveredAt: { type: Date, default: null },
    cancelledAt: { type: Date, default: null },
    cancelRequest: { type: requestStateSchema, default: () => ({}) },
    returnRequest: { type: requestStateSchema, default: () => ({}) },
  },
  { timestamps: true }
);

orderSchema.index({ status: 1 });
orderSchema.index({ createdAt: -1 });
orderSchema.index({ user: 1, createdAt: -1 });
orderSchema.index({ status: 1, deliveredAt: -1 });
orderSchema.index({ 'cancelRequest.status': 1 });
orderSchema.index({ 'returnRequest.status': 1 });
orderSchema.index({ 'items.serialNumber': 1 });

module.exports = mongoose.model('Order', orderSchema);

File: app/src/main/java/com/aurapc/admin/data/model/Order.java
package com.aurapc.admin.data.model;

import com.google.gson.annotations.SerializedName;

import java.util.List;
import java.util.Map;

public class Order {
    public String id;
    @SerializedName("_id")
    public String _id;
    public String orderNumber;
    public String status;
    public String createdAt;
    public String updatedAt;
    public ShippingAddress shippingAddress;
    public List<OrderItem> items;
    public Double subtotal;
    public Double shippingFee;
    public Double discountAmount;
    public Double total;
    public Double totalAmount;
    public String paymentMethod;
    public String paymentStatus;
    public String cancelRequestStatus;
    public String returnRequestStatus;

    public Map<String, Object> user;

    public static class ShippingAddress {
        public String name;
        public String fullName;
        public String phone;
        public String street;
        public String address;
        public String city;
        public String district;
        public String ward;
    }

    public static class OrderItem {
        public String id;
        @SerializedName("_id")
        public String _id;
        public String product;
        public Map<String, Object> productData;
        public String name;
        public String image;
        public Integer qty;
        public Integer quantity;
        public Double price;
        public String serialNumber;
    }
}
